### ISSUE TITLE
If working in Visual Studio 2017

### ISSUE THREAD

**Issue by BuhoSabio** [FranckFreiburger/http-vue-loader#58]
Just to mention, that will need to set mimeType for ".vue" extension file to use httpvueloader

May need to do it in Project "ApplicationHost" config file or in IIS Express "ApplicationHost" config file.

Project:
C:\Users\<your name>\Documents\Visual Studio 2017\Projects\<project name>\.vs\config

IIS Express:
C:\Users\<your name>\Documents\IISExpress\config

<mimeMap fileExtension=".vue" mimeType="application/javascript" /> 

Otherwise, keeps throwing file not found when referencing url file <blah>.vue.

**Comment by giolvani** [comment]
Yeah, this is a common thing to who use visual studio projects...

**Comment by Yxjhasadream** [comment]
Thanks for the tip.
But Chrome often handles Vue files as static resources, so how do I configure config to get the latest modified Vue files correctly every time? I'm using asp.net.

**Comment by Yxjhasadream** [comment]
All right. I found a way.
![image](https://user-images.githubusercontent.com/35222088/53486081-b8ccb280-3ac2-11e9-8967-d58a1986e197.png)

**Comment by giolvani** [comment]
@Yxjhasadream you should use cache-busting on querystring parameters instead use this option from chrome.

**Comment by Yxjhasadream** [comment]
@giolvani  It really should be, thank you. But I usually need chrome not to cache when I'm debugging, and it's better to cache when it's released.

**Comment by BinarySpike** [comment]
In MVC Core projects you can use something like this in your `Startup.cs` Configure:
```
app.UseStaticFiles(new StaticFileOptions
{
    ContentTypeProvider = new FileExtensionContentTypeProvider(
        new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) {
            { ".vue", "application/javascript" }
        }),
});
```

I feel that's more maintainable than in the ApplicationHost.

**Comment by richsoft** [comment]
I know this is an old issue, but another way to set this up in an ASP.NET app, is to add the following to the web.config:

```
  <system.webServer>
    <staticContent>
      <mimeMap fileExtension=".vue" mimeType="application/javascript"/>
    </staticContent>
  </system.webServer>
```

**Comment by SeaLife** [comment]
For some reason the dev server to use in VueJS in a basic VueJS project (started with `npm run serve` which runs `vue-cli-service serve`) `.vue` files are served as XML.
In this case the loader errors out with "responseText is undefined" - i checked networking activities and the result seems to be fine, so i think, this is also mime-type related. Can someone confirm this?
And, can someone provide a fix for this?

**Comment by gaby64** [comment]
Failed to resolve async component default: ReferenceError: responseText is not defined
